src/Compositor.cpp

```diff
@@ -1929,6 +1929,8 @@ void CCompositor::moveWorkspaceToMonitor(CWorkspace* pWorkspace, CMonitor* pMoni
             }
 
             w->updateToplevel();
+
+            g_pProtocolManager->m_pFractionalScaleProtocolManager->setPreferredScaleForSurface(w->m_pWLSurface.wlr(), pMonitor->scale);
```

Comment:
this should be done in updateToplevel, it already has a func for this, but ignores the fractional scale protocol

Reply:
That makes sense.

>it already has a func for this

Do you mean `updateSurfaceOutputs()`?

Since `updateTopLevel` is also called with the `movetoworkspace` dispatcher, I have moved all of the fractional scale code from `moveToWorkspace` (including the part from 70eb74c356d291f4b5e048abef7772864a4f95f1) to `updateSurfaceOutputs`.

Reply:
setpreferredscaleforsurface is a noop if nothing changed so it doesnt matter if it gets called more often than needed tbh